File: legacy/backend/app/domain/models/save.py
from dataclasses import dataclass


@dataclass(frozen=True)
class SaveSummary:
    id: str
    session_id: str
    title: str
    world_name: str
    updated_at: str
    progress: str
    summary: str
    player_character_name: str | None = None
    parent_save_id: str | None = None
    branch_root_save_id: str | None = None
    branch_label: str | None = None

File: src-tauri/src/models/save.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SaveSummary {
    pub id: String,
    pub session_id: String,
    pub title: String,
    pub world_name: String,
    pub updated_at: String,
    pub progress: String,
    pub summary: String,
    pub player_character_name: Option<String>,
    pub parent_save_id: Option<String>,
    pub branch_root_save_id: Option<String>,
    pub branch_label: Option<String>,
    pub turn_index: i32,
}
